src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -4730,12 +4731,13 @@ class MavlinkStreamMountOrientation : public MavlinkStream
 protected:
 	explicit MavlinkStreamMountOrientation(Mavlink *mavlink) : MavlinkStream(mavlink),
 		_mount_orientation_sub(_mavlink->add_orb_subscription(ORB_ID(mount_orientation))),
-		_mount_orientation_time(0)
+		_gpos_sub(_mavlink->add_orb_subscription(ORB_ID(vehicle_global_position)))
 	{}
 
 	bool send(const hrt_abstime t) override
 	{
-		struct mount_orientation_s mount_orientation;
+		mount_orientation_s mount_orientation{};
+		vehicle_global_position_s gpos{};
```

Comment:
You can move this to right above the `_gpos_sub->update(&gpos);` to only bother initializing if mount_orientation has actually updated.

Reply:
fixed this